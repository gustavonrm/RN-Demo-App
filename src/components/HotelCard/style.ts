import { StyleSheet } from 'react-native';
import { GREY, PINK, WHITE } from '../../constants/colors';

const style = StyleSheet.create({
    container: {
        flex: 1,
        borderWidth:1,
        borderColor: GREY,
        borderRadius: 10,
        backgroundColor: WHITE,
        marginVertical: 10,
    },
    image:{
        flex:1,
        width: undefined,
        height: undefined,
        borderRadius: 10,
    },
    priceContainer:{
     position: 'absolute',
       fontWeight: 'bold',
       right: 0,
       bottom: 0,
       margin: 10,
       alignItems: 'flex-end',
    },
    priceText:{
          fontWeight: 'bold',
          color: WHITE,
            fontSize: 24,
          textShadowColor: PINK,
          textShadowOffset: {width: -1, height: 1},
          textShadowRadius: 4,
       },
    priceText2:{
        fontWeight: '100!important',
        color: WHITE,
        fontSize: 14,
        textShadowColor: PINK,
        textShadowOffset: {width: -1, height: 1},
        textShadowRadius: 4,
     },
     priceText3:{
        fontWeight: 'bold',
        color: WHITE,
          fontSize: 14,
        textShadowColor: PINK,
        textShadowOffset: {width: -1, height: 1},
        textShadowRadius: 4,
        marginTop: 5,
     },
    top:{
        borderRadius: 10,
        height: 200,
        flex: 1,
        flexDirection: 'column',
    },
    bottom:{
        flex: 1,
        flexDirection: 'column',
        margin: 10,
    },
});

export default style;
