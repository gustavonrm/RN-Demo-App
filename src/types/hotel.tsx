export interface Hotel {
    id: number,
    name: string,
    location: Location,
    stars: number,
    checkIn: FromToHours,
    checkOut: FromToHours,
    contact: Contact,
    gallery: string[],
    userRating: number,
    price: string,
    currency: string,
}

export interface Location {
    address: string,
    city: string,
    latitude: number,
    longitude: number
}

export interface FromToHours {
    from: string,
    to: string,
}

export interface Contact {
    phoneNumber: string,
    email: string,
}
