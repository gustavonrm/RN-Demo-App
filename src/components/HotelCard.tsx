import React from 'react';
import { Image, Text, TouchableOpacity} from 'react-native';
import { Hotel } from '../types/hotel';
import { useNavigation } from '@react-navigation/native';
import { StackNavigationProp } from '@react-navigation/stack';
import { RootStackParamList } from '../navigationTypes';

type HotelCardProps = {
    hotel: Hotel
}

const HotelCard = ({hotel}: HotelCardProps) => {

const navigation = useNavigation<StackNavigationProp<RootStackParamList>>();
  return (
  <TouchableOpacity onPress={() => navigation.navigate('Preview', {id: hotel.id})}>
    <Image style={{width: 100,height: 100}} source={{uri: hotel.gallery[0] || 'https://upload.wikimedia.org/wikipedia/commons/1/14/No_Image_Available.jpg?20200913095930'}}/>
    <Text>{hotel.name}</Text>
    <Text>{hotel.location.city} - {hotel.location.address}</Text>
    <Text>{hotel.price} {hotel.currency}</Text>
    </TouchableOpacity>
  );
};

export default HotelCard;
