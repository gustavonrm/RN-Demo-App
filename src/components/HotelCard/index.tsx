import React from 'react';
import { Image, Text, TouchableOpacity, View} from 'react-native';
import { Hotel } from '../../types/hotel';
import { useNavigation } from '@react-navigation/native';
import { StackNavigationProp } from '@react-navigation/stack';
import { RootStackParamList } from '../navigationTypes';
import { FontAwesomeIcon } from '@fortawesome/react-native-fontawesome';
import style from './style';

type HotelCardProps = {
    hotel: Hotel
}

const simpleCurrencyFormatter = (price: string, currency: string) => {
  let currencySymbol = currency === 'EUR' ? '€' : '$';
  return currencySymbol + price;
};

const HotelCard = ({hotel}: HotelCardProps) => {

  const navigation = useNavigation<StackNavigationProp<RootStackParamList>>();

  return (
  <TouchableOpacity style={style.container} onPress={() => navigation.navigate('Preview', {id: hotel.id})}>
    <View style={style.top}>
      <Image style={style.image} source={{uri: hotel.gallery[0] || 'https://upload.wikimedia.org/wikipedia/commons/1/14/No_Image_Available.jpg?20200913095930'}}/>
      <View style={style.priceContainer}>
        <Text style={style.priceText}>
          {`${hotel.currency === 'EUR' ? '€' : '$'}${hotel.price}`}<Text style={style.priceText2}>{' /person'}</Text>
        </Text>
        <Text style={style.priceText3}>
        {`${hotel.currency === 'EUR' ? '€' : '$'}${hotel.price}`} {'total'}
        </Text>
      </View>
    </View>
    <View style={style.bottom}>
      <Text>{hotel.location.city} - {hotel.location.address}</Text>
      <Text>{hotel.name}</Text>
      <FontAwesomeIcon icon="star" />
      <Text>{hotel.stars}</Text><Text>{hotel.userRating}</Text>
      <Text>{hotel.checkIn.from}</Text><Text>{hotel.checkIn.to}</Text><Text>{hotel.checkOut.from}</Text><Text>{hotel.checkOut.to}</Text>
    </View>


    </TouchableOpacity>
  );
};

export default HotelCard;
