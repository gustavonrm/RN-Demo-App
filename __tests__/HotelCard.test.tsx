import React from 'react';
import 'react-native';
import { render, screen } from '@testing-library/react-native';
import HotelCard from '../src/components/Hotel/HotelCard';
import { hotel } from './mocks/hotel';

describe('HotelPreview Element', () => {
  it('renders hotel card correctly', () => {
    render(<HotelCard hotel={hotel} />);
    expect(screen.getByTestId('hotelCardTestId')).toBeTruthy();
    expect(screen.getByText(`€${hotel.price} /person`)).toBeTruthy();
    expect(screen.getByText(`€${hotel.price} total`)).toBeTruthy();
    expect(screen.getByText(hotel.name)).toBeTruthy();
    expect(
      screen.getByText(`HOTEL in ${hotel.location.city} - ${hotel.location.address}`)
    ).toBeTruthy();
    expect(screen.getAllByTestId('starTestId').length).toBe(hotel.stars);
    expect(screen.getByText(`| ${hotel.userRating} user rating`)).toBeTruthy();
    expect(
      screen.getByText(`Check in from ${hotel.checkIn.from} to ${hotel.checkIn.to}`)
    ).toBeTruthy();
    expect(
      screen.getByText(`Check out from ${hotel.checkOut.from} to ${hotel.checkOut.to}`)
    ).toBeTruthy();
  });

  it('stars are correctly rendered', () => {
    render(<HotelCard hotel={hotel} />);
    expect(screen.getAllByTestId('starTestId').length).toBe(hotel.stars);
  });
});
