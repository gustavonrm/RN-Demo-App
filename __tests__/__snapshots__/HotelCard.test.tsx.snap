// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HotelPreview Element snapshot test 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "backgroundColor": "#FFF",
      "borderColor": "#DCDCDC",
      "borderRadius": 10,
      "borderWidth": 1,
      "flex": 1,
      "marginVertical": 10,
      "opacity": 1,
      "paddingBottom": 10,
    }
  }
  testID="hotelCardTestId"
>
  <View
    style={
      {
        "borderRadius": 10,
        "flex": 1,
        "flexDirection": "column",
        "height": 200,
      }
    }
  >
    <Image
      source={
        {
          "uri": "https://res.cloudinary.com/lastminute/image/upload/t_OSE_redes_item_view/v1499779963/Swindon_yjsrwz.jpg",
        }
      }
      style={
        {
          "borderRadius": 10,
          "flex": 1,
          "height": undefined,
          "width": undefined,
        }
      }
    />
    <View
      style={
        {
          "alignItems": "flex-end",
          "bottom": 0,
          "fontWeight": "bold",
          "margin": 10,
          "position": "absolute",
          "right": 0,
        }
      }
    >
      <Text
        style={
          {
            "color": "#FFF",
            "fontSize": 24,
            "fontWeight": "bold",
            "textShadowColor": "#E12C70",
            "textShadowOffset": {
              "height": 1,
              "width": -1,
            },
            "textShadowRadius": 4,
          }
        }
      >
        €120
        <Text
          style={
            {
              "color": "#FFF",
              "fontSize": 14,
              "fontWeight": "light !important",
              "textShadowColor": "#E12C70",
              "textShadowOffset": {
                "height": 1,
                "width": -1,
              },
              "textShadowRadius": 4,
            }
          }
        >
           /person
        </Text>
      </Text>
      <Text
        style={
          {
            "color": "#FFF",
            "fontSize": 14,
            "fontWeight": "bold",
            "marginTop": 5,
            "textShadowColor": "#E12C70",
            "textShadowOffset": {
              "height": 1,
              "width": -1,
            },
            "textShadowRadius": 4,
          }
        }
        testID="priceTestId"
      >
        €120
         
        total
      </Text>
    </View>
  </View>
  <View
    style={
      {
        "flex": 1,
        "flexDirection": "column",
        "margin": 10,
      }
    }
  >
    <Text
      numberOfLines={1}
      style={
        {
          "color": "#000",
          "fontSize": 14,
          "fontWeight": "300",
          "marginVertical": 2,
        }
      }
    >
      HOTEL in 
      London
       - 
      6 Hercules Road
    </Text>
    <Text
      style={
        {
          "color": "#000",
          "fontSize": 18,
          "fontWeight": "bold",
          "marginVertical": 2,
        }
      }
    >
      Park Plaza London Waterloo
    </Text>
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
    >
      <View
        style={
          {
            "flexDirection": "row",
          }
        }
      >
        <View
          testID="starTestId"
        />
        <View
          testID="starTestId"
        />
        <View
          testID="starTestId"
        />
        <View
          testID="starTestId"
        />
      </View>
      <Text
        style={
          {
            "color": "#000",
            "fontSize": 14,
            "fontWeight": "500",
            "marginVertical": 4,
          }
        }
      >
         | 
        9.8
         user rating
      </Text>
    </View>
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
    >
      <View />
      <Text
        style={
          {
            "color": "#000",
            "fontSize": 14,
            "fontWeight": "500",
            "marginVertical": 4,
          }
        }
      >
        Check in from
         
        <Text
          style={
            {
              "color": "#000",
              "fontSize": 14,
              "fontWeight": "300",
              "marginVertical": 2,
            }
          }
        >
          14:00
           to 
          20:00
        </Text>
      </Text>
    </View>
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
    >
      <View />
      <Text
        style={
          {
            "color": "#000",
            "fontSize": 14,
            "fontWeight": "500",
            "marginVertical": 4,
          }
        }
      >
        Check out from
         
        <Text
          style={
            {
              "color": "#000",
              "fontSize": 14,
              "fontWeight": "300",
              "marginVertical": 2,
            }
          }
        >
          07:00
           to 
          10:00
        </Text>
      </Text>
    </View>
  </View>
</View>
`;
